controllers/cloud.redhat.com/utils/utils.go

```diff
@@ -254,22 +254,43 @@ func MakePVC(pvc *core.PersistentVolumeClaim, nn types.NamespacedName, labels ma
 	}
 }
 
-// DeploymentStatusChecker takes a deployment and returns True if it is deemed ready by the logic
-// in the function.
+// DeploymentStatusChecker takes a deployment and returns True if it is deemed
+// ready by the logic in the function.
 func DeploymentStatusChecker(deployment *apps.Deployment) bool {
-	if deployment.Generation <= deployment.Status.ObservedGeneration {
-		if deployment.Spec.Replicas != nil && deployment.Status.UpdatedReplicas < *deployment.Spec.Replicas {
-			return false
-		}
-		if deployment.Status.Replicas > deployment.Status.UpdatedReplicas {
-			return false
+	if deployment.Generation > deployment.Status.ObservedGeneration {
+		// The deployment controller still needs to reconcile
+		return false
+	}
+
+	if deployment.Spec.Replicas != nil {
+		if *deployment.Spec.Replicas == 0 {
+			// Since there's no replicas, there's nothing to do to get ready
+			return true
 		}
-		if deployment.Status.AvailableReplicas < deployment.Status.UpdatedReplicas {
+
+		if deployment.Status.UpdatedReplicas < *deployment.Spec.Replicas {
```

Comment:
I'm guessing if `Replicas` is 0, then `UpdatedReplicas` will never be set at all?

Reply:
I haven't tested it, but if there are zero replicas specified, I wouldn't expect there to be any replicas to update.